test/behavior/sniff.test.js

```diff
@@ -111,39 +111,44 @@ test('Should handle hostnames in publish_address', t => {
   })
 })
 
-test('Sniff interval', { skip: 'Flaky on CI' }, t => {
-  t.plan(10)
+test('Sniff interval', t => {
+  t.plan(9)
 
   buildCluster(({ nodes, shutdown, kill }) => {
     const client = new Client({
       node: nodes[Object.keys(nodes)[0]].url,
       sniffInterval: 50
     })
 
+    t.strictEqual(client.connectionPool.size, 1)
+
     // this event will be triggered by api calls
-    client.on(events.SNIFF, (err, request) => {
+    client.on(events.SNIFF, (err, result) => {
       t.error(err)
-      const { hosts, reason } = request.meta.sniff
+      const { hosts, reason } = result.meta.sniff
       t.strictEqual(
         client.connectionPool.size,
         hosts.length
       )
       t.strictEqual(reason, Transport.sniffReasons.SNIFF_INTERVAL)
     })
 
-    t.strictEqual(client.connectionPool.size, 1)
-    setTimeout(() => client.info(t.error), 60)
-
+    // SNIFF event should be emit only when both are met:
+    // 1) delay is greater than sniffInterval
+    // 2) delay is greater than time of last sniff + sniffInterval
+    //
+    // See Transport.js => getConnection(), sniff()
+    setTimeout(() => client.info(), 10)
+    setTimeout(() => client.info(), 20)
+    setTimeout(() => client.info(t.error), 60) // meets 1) and 2)
+    setTimeout(() => client.info(), 80)
+    setTimeout(() => client.info(), 90)
     setTimeout(() => {
-      // let's kill a node
-      kill('node1')
-      client.info(t.error)
+      kill('node1', () => {
+        client.info(t.error) // meets 1) and 2)
+      })
     }, 150)
 
-    setTimeout(() => {
-      t.strictEqual(client.connectionPool.size, 3)
-    }, 200)
-
```

Comment:
Great catch! I would kindly ask you to revert this change, leave the test as it was before and keep just the `kill` fix :)

Reply:
Thanks! Can you please explain to me why to revert the test?

I see 4 reasons why the new test is better. The previous test would _NOT_ work (1), would not be complete (2), can randomly fail (3), and has an unrelated assertion (4).

Here's the detail:

1) By not waiting for `kill()` to finish, `shutdown()` would try to kill 4 nodes instead of 3, because of outdated `nodes`. This would result in the following error:

```
kill('node1')
> kill node1

shutdown()
> kill node0
> kill node1  // Should NOT be here! Cannot kill!
> kill node2
> kill node3

Error [ERR_SERVER_NOT_RUNNING]: Server is not running.
```

2) The previous test does not test **SNIFF** event earlier than `sniffInterval` and earlier than "time of last sniff" + `sniffInterval`.

3) The previous statement:

```js
setTimeout(() => {
  t.strictEqual(client.connectionPool.size, 3)
}, 200)
```

Has an expectation that depends on **SNIFF** to run `this.connectionPool.update(hosts)`. Also depends on lazy loading of `info`, connection speed, time of killing the node.

4) Similar to point 3). This test is primarily about **'Sniff interval'** and not about nodes. Also, `client.on(events.SNIFF, (err, result) => { ... }` already has the assertion needed, and compares 4 nodes with 4, on the first run, and 3 with 3, on the second run.

Please explain. I don't see the reasons to revert the test.